### PR TITLE
refactor: externalize currency tools in a new packages

### PR DESCRIPTION
**Pull request by vrolland**
## Description of the changes

- Move the currency tools to a new package
- Remove depedency to  ̀currency-iso`.

### CODE REVIEW

File: packages/currency/package.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "devDependencies": {
+    "@types/jest": "26.0.13",
+    "@typescript-eslint/parser": "4.1.1",
+    "eslint": "7.9.0",
+    "eslint-plugin-spellcheck": "0.0.17",
+    "eslint-plugin-typescript": "0.14.0",
+    "jest": "26.4.2",
+    "lint-staged": "10.3.0",
+    "mocha": "8.1.3",
+    "nyc": "15.1.0",
```
Comment (benjlevesque) [inline packages/currency/package.json:57]: Those 2 are no longer used. We should remove them from `utils` too, it's a mistake that they're still there.

File: packages/currency/src/iso4217/data-iso4217.ts [hunk 2]
```diff
@@ -0,0 +1,1356 @@
+/*
+	Follows ISO 4217, https://www.iso.org/iso-4217-currency-codes.html
+	See https://www.currency-iso.org/dam/downloads/lists/list_one.xml
+	Data last updated 2018-08-29
+*/
+/* eslint-disable spellcheck/spell-checker */
+export default [
```
Comment (benjlevesque) [inline packages/currency/src/iso4217/data-iso4217.ts:7]: any reason not to no longer use the iso currency package ? would avoid us to require to update this list manually, even if I guess this won't change any time soon :)

File: packages/currency/src/others/others.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export default {
```
Comment (benjlevesque) [inline packages/currency/src/others/others.ts:1]: maybe the folder is unnecessary?  `packages/currency/src/others.ts`

File: packages/currency/src/iso4217/data-iso4217.ts [hunk 4]
```diff
@@ -0,0 +1,1356 @@
+/*
```
Comment (benjlevesque) [inline packages/currency/src/iso4217/data-iso4217.ts:1]: maybe the folder is necessary? `packages/currency/src/iso4217.ts`

**Review by yomarion — APPROVED** [review]
Well done!

**Review by benjlevesque — COMMENTED** [review]
Nice! This will be very useful....

I think you miss the update to the CI (unit tests)

Some suggestions on easier code browsing (most of it comes from previous structure, but it's a good opportunity to do some cleanup) 
- rename `erc20/networks/networks.ts` to `erc20/networks/index.ts`
- rename `erc20/erc20.ts` to `erc20/index.ts`
- delete file extra-data.ts, and move the value to `erc20/networks/mainnet.ts` (this requires some small changes for consistance with the other "custom" currencies)

File: packages/currency/package.json (hunk 1 above)
Comment (vrolland) [reply packages/currency/package.json:57]: @benjlevesque What about the file .nycrc?

File: packages/currency/src/iso4217/data-iso4217.ts (hunk 2 above)
Comment (vrolland) [reply packages/currency/src/iso4217/data-iso4217.ts:7]: As we are getting closer to the protocol this must not change or with a new version of the request logic.
So we cannot depend on an external library.
I don't have the same concern on the ERC20 currencies as the number of decimal is define directly on the smart contract of the token.

**Review by AdamDowson — APPROVED** [review]

File: packages/currency/src/iso4217/data-iso4217.ts (hunk 2 above)
Comment (benjlevesque) [reply packages/currency/src/iso4217/data-iso4217.ts]: fair enough, but that means reviewing this list every now and then. Maybe we should subscribe to changes to the original repo we were using

**Review by benjlevesque — APPROVED** [review]

File: packages/currency/test/currency.test.ts [hunk 5]
```diff
@@ -44,6 +44,7 @@ describe('api/currency', () => {
     });
 
     it('returns INDA', () => {
+      // console.log(getAllSupportedCurrencies().ERC20);
```
Comment (yomarion) [inline packages/currency/test/currency.test.ts:47]: Forgot that?

**Review by yomarion — COMMENTED** [review]
Great job, it can save lives 👍 Isn't it time to give the currency logic an object oriented dev approach? (new Currency('DAI', 'rinkeby')).decimals would help me a lot

**Review by AdamDowson — APPROVED** [review]

**Pull request merged**